File: SILVER/S5_9037_The candy war.py
def check(c):
    if len(set(candy)) == 1:
        return True

    return False


t = int(input())

for _ in range(t):
    n = int(input())
    candy = list(map(int, input().split()))

    cycle = 0
    while True:
        if check(candy):
            break

        next_candy = [0] * n

        for i in range(n):
            if candy[i] % 2:
                candy[i] += 1

        if check(candy):
            break

        for i in range(n):
            next_candy[(i + 1) % n] += candy[i] // 2
            next_candy[i] += candy[i] // 2

        candy = next_candy

        cycle += 1

    print(cycle)

File: SILVER/S5_9037_The candy war.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());

        for (int test = 0; test < t; test++) {
            int n = Integer.parseInt(br.readLine());
            int[] candy = Arrays.stream(br.readLine().split(" "))
                    .mapToInt(Integer::parseInt)
                    .toArray();

            int cycle = 0;
            while (true) {
                if (checkCandy(candy)) {
                    break;
                }

                int[] nextCandy = new int[n];

                for (int i = 0; i < n; i++) {
                    if (candy[i] % 2 != 0) {
                        candy[i]++;
                    }
                }

                if (checkCandy(candy)) {
                    break;
                }

                for (int i = 0; i < n; i++) {
                    nextCandy[(i + 1) % n] += candy[i] / 2;
                    nextCandy[i] += candy[i] / 2;
                }

                candy = nextCandy;
                cycle++;
            }

            System.out.println(cycle);
        }
    }

    private static boolean checkCandy(int[] candy) {
        int first = candy[0];

        for (int c : candy) {
            if (c != first) {
                return false;
            }
        }
        return true;

    }
}